src/CoreWCF.Primitives/src/CoreWCF/Security/FederatedMessageSecurityOverHttp.cs

```diff
@@ -0,0 +1,232 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.ObjectModel;
+using CoreWCF.IdentityModel.Tokens;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+using CoreWCF.Channels;
+using CoreWCF.Security;
+using CoreWCF.Security.Tokens;
+using System.Xml;
+using System.ComponentModel;
+using CoreWCF.Runtime;
+using System;
+
+namespace CoreWCF.Security
+{
+    public sealed class FederatedMessageSecurityOverHttp
+    {
+        internal const bool DefaultNegotiateServiceCredential = true;
+        internal const SecurityKeyType DefaultIssuedKeyType = SecurityKeyType.SymmetricKey;
+        internal const bool DefaultEstablishSecurityContext = true;
+        private SecurityAlgorithmSuite _algorithmSuite;
+        private EndpointAddress _issuerMetadataAddress;
+        private SecurityKeyType _issuedKeyType;
+
+        public FederatedMessageSecurityOverHttp()
+        {
+            NegotiateServiceCredential = DefaultNegotiateServiceCredential;
+            _algorithmSuite = SecurityAlgorithmSuite.Default;
+            _issuedKeyType = DefaultIssuedKeyType;
+            ClaimTypeRequirements = new Collection<ClaimTypeRequirement>();
+            TokenRequestParameters = new Collection<XmlElement>();
+            EstablishSecurityContext = DefaultEstablishSecurityContext;
+        }
+
+        public bool NegotiateServiceCredential { get; set; }
+
+        public SecurityAlgorithmSuite AlgorithmSuite
+        {
+            get { return _algorithmSuite; }
+            set
+            {
+                if (value == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(value));
+                }
+                _algorithmSuite = value;
+            }
+        }
+
+        public bool EstablishSecurityContext { get; set; }
+
+        [DefaultValue(null)]
+        public EndpointAddress IssuerAddress { get; set; }
```

Comment:
You can remove the DefaultValue attributes as they were only there to decide whether to emit the value when writing out config. If it was the default value, it wouldn't be emitted.